subprojects/plugins/src/integTest/groovy/org/gradle/java/compile/AbstractJavaGroovyCompileAvoidanceIntegrationSpec.groovy

```diff
@@ -293,39 +276,55 @@ public class ToolImpl {
                 public Object s = "12";
                 public void execute() { String s = toString(); }
             }
-"""
+        """
 
         then:
         succeeds ":b:${language.compileTaskName}"
         executedAndNotSkipped ":a:${language.compileTaskName}"
         skipped ":b:${language.compileTaskName}"
+    }
 
-        when:
-        // add static initializer and constructor
-        sourceFile.text = """
+    def "doesn't recompile when empty initializer, static initializer or constructor is added"() {
+        given:
+        buildFile << """
+            project(':b') {
+                dependencies {
+                    implementation project(':a')
+                }
+            }
+        """
+        def sourceFile = file("a/src/main/${language.name}/ToolImpl.${language.name}")
+        sourceFile << """
             public class ToolImpl {
-                static { }
-                public ToolImpl() { }
-                public Object s = "12";
-                public void execute() { String s = toString(); }
+                {}
+                static {}
+                public ToolImpl() {}
+                public Object s = String.valueOf(12);
```

Comment:
`s` and the `execute` method are not interesting for this test any more and can be removed. If they are not removed, they probably shouldn't be changed.

Reply:
this is what I saw in the other test as well, but I agree it's better without: removed